intro-dplyr.Rmd

```diff
@@ -0,0 +1,343 @@
+5. The `%>%` pipe operator can help clarify complex data processing workflows.
+
+## Filtering data based on values and ordering data by variables
+
+```{r}
+who6 <- who5 %>%
+  # exclude relapse of TB, include years after 1997 and non-zero case counts
+  filter(type != "rel", year > 1997, cases > 0) %>%
+  # create a new variable called "nordic country"
+  mutate(
+    nordic = if_else(country == "Denmark" | country == "Sweden" | country == "Norway", "Nordic", "not Nordic")
+  ) %>%
+  # arrange data by year and country
+  arrange(year, nordic)
+```
+
+### Exercise: filter and re-arrange your dataset
+
+Time: 15 min
+
+```{r}
+# make a dataset who7 that would include only smear positive TB diagnosed from 1994 through 2012
+who7 <- who5 %>%
+  filter(type == "sp", year >= 1994) %>%
+  # select all variables except for age
+  select(-age) %>%
+  # arrange by number of cases in descending order
+  arrange(desc(cases))
+```
+
+If you are familiar with ggplot2, then you can optional visualize a bar plot of the case count on y-axis and year on x-axis, with sex as the fill.
+
+## Creating a simple data summary
+
+```{r}
+summary <- who6 %>%
+  # group by country
+  group_by(country) %>%
+  # summarise cases count per country
+  summarise(
+    cum_n_cases = sum(cases),
+    pct_fem_cases_cum = sum(cases[sex == "f"]) / cum_n_cases * 100,
+    min_cases = min(cases),
+    max_cases = max(cases)
+  ) %>%
+  arrange(desc(cum_n_cases))
+summary
+```
+
+## Converting to long form with `gather()`
+
+Explore the structure of the `table4a`.
+
+```{r}
+table4a
+```
+
+- Is `table4a` tidy?
+- How would you like to organise data in `table4`?
+    - Would it be helpful to have columns `country`, `year`, and `cases` with counts?
+    - Gather columns `1999` and `2000` into row values of a new column `year` and cases counts into values of a new column `cases`
+
+```{r}
+table4a_tidy <- table4a %>%
+  gather(key = "year", value = "cases", `1999`:`2000`)
+table4a_tidy
+```
+
+### Exercise: Convert your data into a long form
+
+Time: 5 min
+
+```{r}
+# explore structure of `table4b`
+table4b
+
+# use gather() to make it tidy
+table4b_tidy <- table4b %>%
+  gather(key = "year", value = "populaton", `1999`, `2000`)
+
+table4b_tidy
+```
+
+## Converting to wide form with `spread()`
+
+Explore the structure of the `table2`
+
+```{r}
+table2
+```
+
+- Is `table2` tidy?
+    - Depends on what we plan to use it for.
+    - Later we would like to calculate the rate of the disease in each country per year: `rate = cases / 100,000 * population`.
+
+- What data structure is preferred?
+    - Each country has a row that contains all information needed to proceed with the calculation: year, number of cases, and the population size.
+    - Spread the column `type` into two columns `cases` and `population`.
+    
+```{r}
+table2 %>%
+    spread(key = type, value = count)
+```
+
+## Case study for using tidyr
```

Comment:
Here, I would use the same dataset throughout rather than use as a case example. Have the final exercise as the "case example"